### ISSUE TITLE
`order-in-character-class`: Reported on references

### ISSUE THREAD

**Issue by fisker** [ota-meshi/eslint-plugin-regexp#231]
```js
const jsxWhitespaceChars = " \n\r\t";
const matchJsxWhitespaceRegex = new RegExp("([" + jsxWhitespaceChars + "]+)");
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ The problem is reported here
```

The location is miss leading, because there is no `\n` there.

Is it hard? I didn't read code, but I think strings are concated, lost original node info?

Another problem `\n`/`\t`/` ` should be escaped in message, they are invisible.

**Comment by fisker** [comment]
Or maybe we can just display the Regexp literal value in message.

**Comment by fisker** [comment]
`no-useless-non-capturing-group` rule have the same issue.
